tabular/src/autogluon/tabular/learner/abstract_learner.py

```diff
@@ -138,6 +140,23 @@ def predict(self, X: DataFrame, model=None, as_pandas=False):
     def _validate_fit_input(self, X: DataFrame, **kwargs):
         if self.label not in X.columns:
             raise KeyError(f"Label column '{self.label}' is missing from training data. Training data columns: {list(X.columns)}")
+        self.sample_weight = kwargs.get('sample_weight', None)
+        self.weight_evaluation = kwargs.get('weight_evaluation', False)
+        X_val = kwargs.get('X_val', None)
+        if self.sample_weight is not None:
+            if not isinstance(self.sample_weight, str):
+                raise ValueError("sample_weight must be a string indicating the name of column that contains sample weights. If you have a vector of sample weights, first add these as an extra column to your data.")
+            if self.sample_weight not in X.columns:
+                raise KeyError(f"sample_weight column '{self.sample_weight}' is missing from training data. Training data columns: {list(X.columns)}")
+            weight_vals = X[self.sample_weight]
+            if weight_vals.isna().sum() > 0:
```

Comment:
given how many checks we have here, probably good to move to its own private method to make code cleaner.